framework/test/unit/modules/interoperability/mainchain/commands/message_recovery.spec.ts

```diff
@@ -0,0 +1,293 @@
+	let transaction: Transaction;
+	let transactionParams: MessageRecoveryParams;
+	let encodedTransactionParams: Buffer;
+	let transactionContext: TransactionContext;
+	let storeMock: StoreMock;
+	let ccms: CCMsg[];
+
+	beforeEach(() => {
+		interoperableCCAPIs = new Map();
+		ccCommands = new Map();
+
+		messageRecoveryCommand = new MessageRecoveryCommand(
+			MODULE_ID_INTEROPERABILITY,
+			interoperableCCAPIs,
+			ccCommands,
+		);
+
+		ccms = [
+			{
+				nonce: BigInt(0),
+				moduleID: 1,
+				crossChainCommandID: 1,
+				sendingChainID: 2,
+				receivingChainID: 3,
+				fee: BigInt(1),
+				status: 1,
+				params: Buffer.alloc(0),
+			},
+		];
+
+		const ccmsEncoded = ccms.map(ccm => codec.encode(ccmSchema, ccm));
+
+		transactionParams = {
+			chainID: 3,
+			crossChainMessages: [...ccmsEncoded],
+			idxs: [0],
+			siblingHashes: [getRandomBytes(32)],
+		};
+
+		encodedTransactionParams = codec.encode(messageRecoveryParams, transactionParams);
+
+		transaction = new Transaction({
+			moduleID: MODULE_ID_INTEROPERABILITY,
+			commandID: COMMAND_ID_MESSAGE_RECOVERY,
+			fee: BigInt(100000000),
+			nonce: BigInt(0),
+			params: encodedTransactionParams,
+			senderPublicKey: getRandomBytes(32),
+			signatures: [],
+		});
+
+		transactionContext = createTransactionContext({
+			transaction,
+		});
+
+		commandExecuteContext = transactionContext.createCommandExecuteContext<MessageRecoveryParams>(
+			messageRecoveryParams,
+		);
+
+		storeMock = {
+			addToOutbox: jest.fn(),
+			getChainAccount: jest.fn(),
+			getTerminatedOutboxAccount: jest.fn(),
+			setTerminatedOutboxAccount: jest.fn(),
+			chainAccountExist: jest.fn().mockResolvedValue(true),
+			isLive: jest.fn().mockResolvedValue(true),
+			terminatedOutboxAccountExist: jest.fn().mockResolvedValue(true),
+		};
+
+		jest
+			.spyOn(messageRecoveryCommand, 'getInteroperabilityStore' as any)
+			.mockImplementation(() => storeMock);
+		jest.spyOn(regularMerkleTree, 'calculateRootFromUpdateData').mockReturnValue(Buffer.alloc(32));
+
+		let chainID;
+		for (const ccm of ccms) {
+			chainID = getIDAsKeyForStore(ccm.sendingChainID);
+
+			when(storeMock.getChainAccount)
+				.calledWith(chainID)
+				.mockResolvedValue({
+					name: `chain${chainID.toString('hex')}`,
+					status: CHAIN_ACTIVE,
+					networkID,
+					lastCertificate: {
+						height: 1,
+						timestamp: 10,
+						stateRoot: Buffer.alloc(0),
+						validatorsHash: Buffer.alloc(0),
+					},
+				});
+		}
+
+		chainID = getIDAsKeyForStore(transactionParams.chainID);
+
+		when(storeMock.getTerminatedOutboxAccount)
+			.calledWith(chainID)
+			.mockResolvedValue({
+				outboxRoot: getRandomBytes(32),
+				outboxSize: 1,
+				partnerChainInboxSize: 1,
+			});
+	});
+
+	it('should successfully process recovery transaction', async () => {
+		// Global Setup
+		await messageRecoveryCommand.execute(commandExecuteContext);
+		expect.assertions(ccms.length + 1);
+
+		{
+			// Assign & Arrange
```

Comment:
maybe `// Arrange` is enough, following `// Arrange // Act // Assert`

Reply:
👍